tests/test_models.py

```diff
@@ -1047,3 +1048,81 @@ def test_eqtransformer_annotate_window_post():
     assert np.isnan(blinded[:100]).all()
     assert (blinded[100:900] == 1).all()
     assert np.isnan(blinded[900:]).all()
+
+
+def test_model_local_saving(tmp_path):
```

Comment:
Good point I will add these in. I've gone about trying to account for parameter overwritten with a slightly tweaked approach, so if your happy with this method, I'll continue with the unit tests, but it would be worth checking the logic first.